testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/SAMLClockSkewAdapterTest.java

```diff
@@ -0,0 +1,145 @@
+@AppServerContainer(ContainerConstants.APP_SERVER_EAP6)
+@AppServerContainer(ContainerConstants.APP_SERVER_EAP71)
+@AppServerContainer(ContainerConstants.APP_SERVER_TOMCAT7)
+@AppServerContainer(ContainerConstants.APP_SERVER_TOMCAT8)
+@AppServerContainer(ContainerConstants.APP_SERVER_TOMCAT9)
+@AppServerContainer(ContainerConstants.APP_SERVER_JETTY92)
+@AppServerContainer(ContainerConstants.APP_SERVER_JETTY93)
+@AppServerContainer(ContainerConstants.APP_SERVER_JETTY94)
+public class SAMLClockSkewAdapterTest extends AbstractServletsAdapterTest {
+
+    private static final String CONTEXT_ROOT = "sales-post-clock-skew";
+    private static final String DEPLOYMENT_NAME_3_SEC = CONTEXT_ROOT + "_3Sec";
+    private static final String DEPLOYMENT_NAME_30_SEC = CONTEXT_ROOT + "_30Sec";
+
+    @ArquillianResource private Deployer deployer;
+
+    @Deployment(name = DEPLOYMENT_NAME_3_SEC, managed = false)
+    protected static WebArchive salesPostClockSkewServlet3Sec() {
+        return samlServletDeployment(CONTEXT_ROOT, DEPLOYMENT_NAME_3_SEC, CONTEXT_ROOT + "/WEB-INF/web.xml", 3, AdapterActionsFilter.class, PublicKeyLocator.class, SendUsernameServlet.class);
+    }
+    @Deployment(name = DEPLOYMENT_NAME_30_SEC, managed = false)
+    protected static WebArchive salesPostClockSkewServlet30Sec() {
+        return samlServletDeployment(CONTEXT_ROOT, DEPLOYMENT_NAME_30_SEC, CONTEXT_ROOT + "/WEB-INF/web.xml", 30, AdapterActionsFilter.class, PublicKeyLocator.class, SendUsernameServlet.class);
+    }
+
+    @Override
+    public void addAdapterTestRealms(List<RealmRepresentation> testRealms) {
+        testRealms.add(IOUtil.loadRealm("/adapter-test/keycloak-saml/testsaml.json"));
+    }
+
+    private String getClockSkewSerlverUrl() throws Exception {
+        return AppServerTestEnricher.getAppServerContextRoot() + "/" + CONTEXT_ROOT;
+    }
+
+    private void setAdapterAndServerTimeOffset(int timeOffset) throws Exception {
+        super.setAdapterAndServerTimeOffset(timeOffset, getClockSkewSerlverUrl() + "/unsecured");
+        DroneUtils.getCurrentDriver().navigate().to(getClockSkewSerlverUrl());
+    }
+
+    private void assertOutcome(int timeOffset, Matcher matcher) throws Exception {
+        try {
+            String resultPage = new SamlClientBuilder()
+                    .navigateTo(getClockSkewSerlverUrl())
+                    .processSamlResponse(POST).build()
+                    .login().user(bburkeUser).build()
+                    .processSamlResponse(POST)
+                    .transformDocument(doc -> {
+                        setAdapterAndServerTimeOffset(timeOffset);
+                        return doc;
+                    }).build().executeAndTransform(resp -> EntityUtils.toString(resp.getEntity()));
+
+            Assert.assertThat(resultPage, matcher);
+        } finally {
+            setAdapterAndServerTimeOffset(0);
+        }
+    }
+
+    @Test
+    public void testErrorExample() throws Exception {
```

Comment:
nit: the test method name was meant to be changed, I was sending it as an example for you to make it easier for you add another tests.